landoapi/models/landing_job.py

```diff
@@ -132,7 +132,7 @@ class LandingJob(Base):
     # Identifier of the published commit which this job should land on top of.
     target_commit_hash = db.Column(db.Text(), nullable=True)
 
-    revisions = db.relationship(
+    revisions: list[Revision] = db.relationship(
```

Comment:
nit: `InstrumentedList` would be more accurate here? Although The type is also ambiguous (could be `RelationshipProperty`, `InstrumentedAttribute`, or `InstrumentedList` depending on how you're looking at this object).

What benefit does adding this type hint provide us, and why just here and nowhere else?

Reply:
The issue right now is that using language features on any subfields of revision don't work, because the LSP can't infer that `revisions` is a list of `Revision` objects due to the dynamic nature of `db.relationship`. Adding this allows using those features on the `revisions` field. For example, try using "get all references" on `Revision.revision_id`, it won't return the usage in `LandingWorker.run_job`. Running it on that usage won't return any references either. 

We could add type hints to the other fields too, but this specific type hint adds tooling features, so I made a point of adding it.

Reply:
Also, we could make the type hint reflect the actual ORM type, but since we don't typically use any of those features I think `list[Revision]` is fine (that's really how we think of the `revisions` attribute anyway). We're planning to tear all this out soon enough, so having this in the short term while we're still on Flask-Lando seems reasonable.